File: j2k-eval/edge-cases/src/NestedAnonymousClasses.java
package edgecases;
import java.util.Comparator;
import java.util.concurrent.Callable;
public class NestedAnonymousClasses {
    public Callable<Comparator<String>> createNestedAnonymous() {
        return new Callable<Comparator<String>>() {
            @Override public Comparator<String> call() {
                return new Comparator<String>() {
                    @Override public int compare(String a, String b) { return a.length() - b.length(); }
                };
            }
        };
    }
    public Runnable createWithCapture(final String prefix) {
        final int[] counter = {0};
        return new Runnable() {
            @Override public void run() { counter[0]++; System.out.println(prefix + ": " + counter[0]); }
        };
    }
}

File: j2k-eval/edge-cases/converted/NestedAnonymousClasses.kt
package edgecases

/**
 * Edge case: Nested anonymous classes.
 * Hypothesis: The converter will struggle with deeply nested anonymous class hierarchies.
 */
class NestedAnonymousClasses {
    fun createNestedAnonymous(): java.util.concurrent.Callable<java.util.Comparator<String>> {
        return object : java.util.concurrent.Callable<java.util.Comparator<String?>?> {
            override fun call(): java.util.Comparator<String> {
                return object : java.util.Comparator<String?> {
                    override fun compare(a: String, b: String): Int {
                        return a.length - b.length
                    }
                }
            }
        }
    }

    fun createWithCapture(prefix: String): java.lang.Runnable {
        val counter: IntArray = intArrayOf(0)
        return object : java.lang.Runnable {
            override fun run() {
                counter.get(0)++
                println(prefix + ": " + counter.get(0))
            }
        }
    }
}